File: my-workout-api/src/main/java/com/avernet/myworkoutapi/exercise/MuscleGroupExercises.java
package com.avernet.myworkoutapi.exercise;

import com.avernet.myworkoutapi.muscle.Muscle;
import com.avernet.myworkoutapi.musclegroup.MuscleGroup;

import java.util.List;

public record MuscleGroupExercises(MuscleGroup muscleGroup, List<ExerciseAddedToWorkout> exerciseAddedToWorkouts, List<Muscle> muscles) {
}

File: my-workout-app/src/interfaces/MuscleGroupExercises.ts
import { MuscleGroup } from './MuscleGroup';
import { ExerciseAddedToWorkout } from './ExerciseAddedToWorkout';
import { Muscle } from './Muscle';

export class MuscleGroupExercises {
    muscleGroup: MuscleGroup;
    exerciseAddedToWorkouts: ExerciseAddedToWorkout[];
    muscles: Muscle[];
}
